sw/device/lib/dif/dif_alert_handler.h

```diff
@@ -0,0 +1,699 @@
+    bool *can_clear);
+
+/**
+ * Disables escalation clearing for this class.
+ *
+ * This operation is similar to locking in that it cannot be undone.
+ *
+ * @param handler An alert handler handle.
+ * @param alert_class The class to disable clearing for.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_alert_handler_result_t dif_alert_handler_escalation_disable_clearing(
+    const dif_alert_handler_t *handler, dif_alert_handler_class_t alert_class);
+
+/**
+ * Clears an on-going escalation, as well as the class accumulator.
+ *
+ * This operation can be disabled with
+ * `dif_alert_handler_escalation_disable_clearing()`.
+ *
+ * @param handler An alert handler handle.
+ * @param alert_class The class to clear an escalation for.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_alert_handler_result_t dif_alert_handler_escalation_clear(
+    const dif_alert_handler_t *handler, dif_alert_handler_class_t alert_class);
+
+/**
+ * Gets the accumulator value for this class.
+ *
+ * This value is the number of alerts of this class that have been logged so
+ * far (more or less, since multiple alerts on the same cycle will be merged
+ * into one). Once this value equals the configured threshold, any followup
+ * alerts will immediately trigger the escalation protocol.
+ *
+ * This value is cleared as a side-effect of
+ * `dif_alert_handler_escalation_clear()`.
+ *
+ * @param handler An alert handler handle.
+ * @param alert_class The class to get the accumulator for.
+ * @param alerts Out-param for the accumulator.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_alert_handler_result_t dif_alert_handler_get_accumulator(
+    const dif_alert_handler_t *handler, dif_alert_handler_class_t alert_class,
+    uint16_t *alerts);
+
+/**
+ * Gets the current value of the "escalation counter".
+ *
+ * The interpretation of this value depends on the value returned by
+ * `dif_alert_handler_class_state_get()`. If it is in the timeout state,
+ * it returns the number of cycles counted towards that cycle so far.
+ * If in an escalation phase, it returns the number of cycles that phase
+ * has been active for.
+ *
+ * @param handler An alert handler handle.
+ * @param alert_class The class to set the counter for.
+ * @param cycles Out-param for the counter.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_alert_handler_result_t dif_alert_handler_get_escalation_counter(
+    const dif_alert_handler_t *handler, dif_alert_handler_class_t alert_class,
+    uint32_t *cycles);
+
+/**
+ * Gets the current state of this class.
+ *
+ * See `dif_alert_handler_class_state_t` for potential states.
+ *
+ * @param handler An alert handler handle.
+ * @param alert_class The class to get the state of
+ * @param state Out-param for the class state.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_alert_handler_result_t dif_alert_handler_get_lass_state(
```

Comment:
```suggestion
dif_alert_handler_result_t dif_alert_handler_get_last_state(
```

Reply:
Actually, that's supposed to be "class". Fixed, at any rate.